libnymea-mqtt/mqttserver.cpp

```diff
@@ -231,10 +231,9 @@ void MqttServerPrivate::onClientConnected(MqttServerClient *client)
     // Start a 10 second timer to clean up the connection if we don't get data until then.
     QTimer *timeoutTimer = new QTimer(this);
     connect(timeoutTimer, &QTimer::timeout, client, [this, client]() {
-        qCWarning(dbgServer) << "A client connected but did not send data in 10 seconds. Dropping connection.";
-        client->abort();
+        qCWarning(dbgServer) << "A client connected but did not send data in 10 seconds. Dropping connection from" << client->peerAddress();
         pendingConnections.take(client)->deleteLater();
-        client->deleteLater();
```

Comment:
I assume abort is triggering an other cleanup?

Reply:
yes. The abort will cause a disconnected signal to be emitted, which will do a cleanup of the client, also deleting it. That caused a "Unexpected null receiver" warning as the deleteLater() would be scheduled on an object which is deleted by that time.